beacon-chain/operations/attestations/prune_expired.go

```diff
@@ -35,6 +35,9 @@ func (s *Service) pruneExpiredAtts() {
 		}
 	}
 
+	if _, err := s.pool.DeleteSeenUnaggregatedAttestations(); err == nil {
```

Comment:
`UnaggregatedAttestations()` is also used in other places. We should be applying `DeleteSeenUnaggregatedAttestations()` to those places as well to rule out any side effects

Reply:
Well, when it comes to returned list of attestations - then the very same lists will be returned (the only thing that is changed is side-effect of removing seen attestation is not applied, but seen attestations are not returned).
The `DeleteSeenUnaggregatedAttestations()` is called on a regular basis in `prune_expired.go`, which should be enough?

Reply:
Say the following timeline:

1. prune period
2. unaggregated_att_1 gets inserted in the pool
3. duplicated unaggregated_att_1 gets inserted in the pool
4. proposer retrieves unaggregated atts in the pool. (Wouldn't duplicated be used here?)
5. prune period

Reply:
`UnaggregatedAttestations()` will not return duplicates -- they are handled just as before, I just do not delete them at the end. Please, see:
https://github.com/prysmaticlabs/prysm/pull/7204/files#diff-9d7755c9f62602ef4e1eba110d6873e9R60-R61

Or am I missing sth?

Reply:
![image](https://user-images.githubusercontent.com/188194/92756364-71043800-f395-11ea-8399-edd795f5bfe1.png)

Reply:
Ah i see. Agree this is fine then. The only down side is 1 day someone could change prune period to something longer without understanding the implications, but that's out of the scope of this concern. Approving!

Reply:
Thank you!